course5/homework04/homework04.cc

```diff
@@ -0,0 +1,122 @@
+#include <iostream>
+#include <cstdio>
+#include <cstdlib>
+#include <stdexcept>
+#include <string>
+
+/***
+Define a “Stream” class, that allows for easy printing of values to a file
+- Let the class manage the file: ie open the file in constructor and close it in destructor.
+- Overload operator<< for the class, so that it can be used to print some types.
+- Define a global variable of the “Stream” type, call it, say, “Cout”, that will write to stdout.
+***/
+
+class Stream
+{
+public:
+	// Constructors
+	Stream() // default
+	{
+		throw std::runtime_error("Stream must be constructed with a filename: Stream('filename.txt')");
+	}
+	Stream(const Stream&) // copy
+	{
+		throw std::runtime_error("Stream object cannot be copied");
```

Comment:
I decided copy assignment should be ok, because the old object ceases to exist (right?), but I don't think it's clear what it would mean for a Stream itself to be copied (Does it stream to the same file as the original object? Does it copy the file to "filename-copy" and start streaming to that one?) So I opted for the error 😃